specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -84,38 +84,48 @@ Power Management compliance.
   * It will subscribe to a In-memory Push event handler hanging off the retrieved Push Channel component.
     * The app should ideally handle the event by setting the Handled property to true to prevent Background Activation from <br/>
     launching a new process.
-
+* If the application is unpackaged, it will use the ProtocolActivator to register the activator instead. The app will also need to 
+call into the bootstrapper API to have access to the Windows App SDK. Click [here](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/tutorial-unpackaged-deployment?tabs=cpp) for information on how to build and 
+deploy unpackaged applications with Windows App SDK.
 ```cpp
 int main()
 {
-    // Register the COM Activator GUID
-    PushNotificationActivationInfo info(
-        PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+    PushNotificationActivationInfo info = nullptr;
+    bool isComActivationSupported = PushNotificationManager::IsActivatorSupported(PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator);
+    if(isComActivationSupported)
+    {
+        // Register the ComActivator guid
+        info = PushNotificationActivationInfo(PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator,
```

Comment:
Is there another model instead here, like:

```
auto info = PushNotificationActivationInfo(PushNotificationRegistrationOptions::Default);
```

... where "default" triggers this "is supported" thing, but _inside_ the activation constructor? Or is there a clear reason to make the app developer have this in their application?

Reply:
Because we support out-of-proc activation for packaged apps, they are allowed to register only a PushTrigger for process A and ComActivator for process B. I think default would be confusing for this case. Also, making the developer know exactly which flag is being used for RegisterActivator will allow them to know which flag to Unregister with.